File: src/main/java/org/gangel/graphomance/usecases/node/CreateSingleVertexLongUniqueHashIndex.java
package org.gangel.graphomance.usecases.node;

import org.gangel.graphomance.DbType;
import org.gangel.graphomance.IndexType;

public class CreateSingleVertexLongUniqueHashIndex extends CreateSingleVertexBase {

    public CreateSingleVertexLongUniqueHashIndex() {
        super("Create node with long unique hash index", IndexType.HASH_UNIQUE, true, false);
    }

    @Override
    public boolean skipFor(DbType dbType) {
        return (dbType == DbType.NEO4J);
    }
}

File: src/main/kotlin/org/gangel/graphomance/usecases/node/CreateSingleVertexLongUniqueHashIndex.kt
package org.gangel.graphomance.usecases.node

import org.gangel.graphomance.DbType
import org.gangel.graphomance.IndexType

class CreateSingleVertexLongUniqueHashIndex :
	CreateSingleVertexBase("Create node with long unique hash index",
						   IndexType.HASH_UNIQUE,
						   true,
						   false) {
	override fun skipFor(dbType: DbType): Boolean {
		return dbType == DbType.NEO4J
	}
}